### ISSUE TITLE
Option to easily disable some control, like in insomnia 

### ISSUE THREAD

**Issue by maurodesouza** [storybookjs/storybook#16909]
**Is your feature request related to a problem? Please describe**

I'm always frustrated when i have some props that receives some array or object, and based on those props render something, and when i will show it to someone or test this conditional render, i need to delete the control value or create a custom control to take care of it, something like this:

```ts
import { Component, Types } from '.'

type ComponentStoryTypes = Types & {
  showSomeArray: boolean
  showActionButton: boolean
}

const someArray= [/* datas... */]

const actionButton = { label: 'Buy now'. action: () => doSomething() }

export default {
  title: 'Component',
  component: Component,
  args: {
    someArray,
    showSomeArray: true // This prop dont belongs to the component

    actionButton,
    showActionButton: true // This prop dont belongs to the component
  }
} as Meta<ComponentStoryTypes >
```

And then manipulate the rendering of someArray and actionButton prop manually with showSomeArray and showActionButton in story rendering to simulate an empty array or the actionButton not being passed

**Describe the solution you'd like**
Some checkbox, when clicked, the control be as disabled, but with its value filled, and when clicking again, the control will be enable again.

When its disabled, the control's value will not be passed to the story, as if the control were empty

**Describe alternatives you've considered**
Some like in insomnia

https://user-images.githubusercontent.com/54520907/144736652-2cf32200-7c56-4b95-bb4f-96a720476ab7.mp4

**Are you able to assist to bring the feature to reality?**
I would be very happy to create this functionality if it doesn't already exist!

Sorry by my english 😅

**Comment by shilman** [comment]
@domyen @tmeasday WDYT? we updated the UI for undefined values in 6.2 but there's currently no way to unset a value. This is one possible UI for that.

**Comment by tmeasday** [comment]
I think it'd make sense for optional values.

PS @maurodesouza in 6.5 we are experimenting with the idea of arg "targets" (where the component/story is only the default target). We shipped this PR in 6.4 to allow experimentation: https://github.com/storybookjs/storybook/pull/16333

I'm not sure it solves the problem for you but it would do half of it (i.e. stop you from having to not pass the `showSomeArray` arg through as a prop). Let me know if you have any thought son that.
